src/pybind/mgr/dashboard/frontend/src/app/shared/services/combo-box.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ComboBoxService {
+  private searchSubject = new Subject<any>();
+
+  constructor() {}
+
+  emit(value: any) {
+    this.searchSubject.next(value);
+  }
+
+  subscribe<T>(handler: (value: T) => void, debounceMs = 300) {
+    this.searchSubject
+      .pipe(
+        debounceTime(debounceMs),
+        distinctUntilChanged((prev, curr) => JSON.stringify(prev) === JSON.stringify(curr))
```

Comment:
```suggestion
        distinctUntilChanged()
```
What are the types ? cant this be just this ?
I am concerned over using stringify as for nested objects it will get inefficient.
Can you add types at all places to understand better, thanks.

Reply:
good catch. this was something i wrote for my local testing which I eventually abandoned but forgot to remove. removed it now.